src/main/webapp/mapScript.js

```diff
@@ -15,37 +15,117 @@
     },
   });
 
-  // Set markers on the map if marker flag is True.
-  if (showMarkers) {
-    let markerJapanTemple = new google.maps.Marker({
-      position: japanTemple,
-      map: map,
-    });
-    let markerNewZealand = new google.maps.Marker({
-      position: newZealandLake,
-      map: map,
-    });
-    let markerUganda = new google.maps.Marker({
-      position: ugandaView,
+  map.addListener("click", function (event) {
+    placeMarker(event.latLng);
+  });
+
+  console.log("Initialise new map");
+
+  function placeMarker(location) {
+    let marker = new google.maps.Marker({
+      position: location,
       map: map,
     });
+
+    let lat = location.lat(),
+      lng = location.lng();
+
+    let data = { lat, lng };
+    console.log("data");
+    console.log(data);
+    let options = {
+      method: "POST",
+      body: JSON.stringify(data),
+      headers: {
+        "Content-Type": "application/json",
+      },
+    };
+
+    fetch("/markers", options)
+      .then((response) => response.json())
+      .then((id) => {
+        console.log("receive new place's id");
+        console.log(id);
+
+        addNewTableItem("New place name", id.toString());
+
+        console.log("Place user's marker");
+      });
+    console.log("Store new marker");
   }
-  console.log("Initialise new map");
+}
+
+function loadRoute() {
+  let options = {
+    method: "GET",
+  };
+
+  fetch("/markers", options)
+    .then((response) => response.json())
+    .then((placesList) => {
+      console.log("Get json of places");
+
+      for (i in placesList) {
+        console.log("Place " + i.toString());
+        console.log(placesList[i].id);
+        let lat = placesList[i].lat,
+          lng = placesList[i].lng,
+          id = placesList[i].id;
+        let position = { lat, lng };
+        let marker = new google.maps.Marker({
+          position: position,
+          map: map,
+        });
+
+        addNewTableItem("Place name " + i.toString() + " ", id.toString());
+      }
+
+      console.log("Place markers");
+    });
+}
+
+function addNewTableItem(name, id) {
+  let tabel = document.getElementById("places-table");
+
+  let newPlace = document.createElement("li");
+  newPlace.id = id;
+  newPlace.classList.add("new-place");
+
+  let markerSign = '<span class="fas fa-ellipsis-v"></span>';
+  let placeName = name;
+  let deleteFunction = "deletePlace('" + id + "')";
+  let deleteSign =
+    '<span class="fas fa-minus-square" onclick="' +
+    deleteFunction +
+    '"></span>';
+
+  newPlace.innerHTML = markerSign + placeName + deleteSign;
+
+  tabel.appendChild(newPlace);
+}
+
+function deletePlace(contentId) {
+  console.log(contentId);
+  let tableItem = document.getElementById(contentId);
+  tableItem.parentNode.removeChild(tableItem);
+  console.log("Remove place from the table");
+
+  let options = {
+    method: "POST",
+  };
+
+  let URL = "/delete_marker?contentId=" + contentId;
+  fetch(URL, options).then();
```

Comment:
I think you should add a   `location.reload();` to the end of that function so that the markers would actually disappear from the map.